ONSdigital/response-operations-ui
# Us103 add initial mandatory dates for ce

# Motivation and Context


This allows internal user to add initial event dates for a collection exercise. This helps to fill the gap between creating and executing a collection exercise by an internal user. This is feature flagged until functionality for creation of action rules based on creating and editing these event dates is completed.

# What has changed

Added 2 endpoints to collection exercise blueprint in order to create new collection exercise event in collection exercise service and added the controller for this. Added confirmation view for creation of a collection exercise event.

Refactored the way errors are being handled by the collection exercise details view to a consistent error handling.

Tests covering errors in the create collection event process have been written e.g. entering an invalid date. Tests have been written to cover communications with the API.

# How to test?

Manual test: View a collection exercise with no or missing event dates add an event date using the add button. Should be returned to the collection exercise details page where the new date appears.



# Links

[Trello] (https://trello.com/c/ZPLXdgU8)
[GitHub Issue] https://github.com/ONSdigital/response-operations-ui/issues/158


PR:
https://github.com/ONSdigital/rasrm-acceptance-tests/pull/142

# Screenshots (if appropriate):
